Feature/#162 semantic tags for header and footer

Ibinola opened the pull request:
---
name: Pull Request Template
about: Use semantic tags for header and footer
title: ''
labels: ''
assignees: ''

---

What is this PR about?
Story:
As a user I would like to see the header and footer components use semantically correct tags for accessibility.

Details:
Rather than using a plain div tag, the header can use header and footer can use footer.

Resolves #162

Ibinola committed:
Replaced <div> tag with <header> tag

Rather than using a plain div tag, the header tag is now used which is semantically correct for accessiblity

Resolves #162

Ibinola committed:
Replaced <div> tag with <footer> tag

Rather than using a plain <div> tag, the <footer> tag is used which is semantically correct for accessibility

Resolves #162

Ibinola committed:
Replaced <div> tag with <footer> tag

Rather than using a plain <div> tag, the <footer> tag is used which is semantically correct for accessibility

Resolves #162

jaytula reviewed (changes requested):
Hmmm... it doesn't look like we actual use the `Footer` component anywhere though it is imported in `Layout`. As far is the changes in PR, I see the use of two `<footer>` within `Footer`. I believe it is better to just have one wrap all the content that is considered the footer.

Ibinola commented:
okay, would fix that

Ibinola committed:
Fix footer

Removed one footer tag, and wrapped all the content considered the footer.

Ibinola committed:
Merge branch 'feature/#162_semantic-tags-for-header-and-footer' of https://github.com/Ibinola/moonshot-chingu-quiz into feature/#162_semantic-tags-for-header-and-footer

One footer tag used to wrap the Footer component

Ibinola committed:
Removed the unused import of 'Footer'

The imported module wasn't used

ZumDeWald commented on components/layout/Footer.tsx:
```diff
@@ -7,7 +7,7 @@ const Footer = () => {
     <FooterWrapper>
       <ContentWrapper>
-        <div>
+        <footer>
```
@Ibinola Instead of adding a footer tag here, inside the wrappers, I think it might be better if this whole thing was properly tagged as a Footer.

To do that you would need to change the base element type for the `FooterWrapper`, that happens in the `components > layout > styles.ts` file on line 215. 
There the `FooterWrapper` is being based based on a `div`, if you change that `div` to `footer` it will make the `FooterWrapper` render out as a Footer.

Then here on line 10 would have to go back to a regular `div`.

ZumDeWald commented on components/layout/Layout.tsx:
```diff
@@ -5,7 +5,6 @@ import React, { useState, useEffect, Fragment } from "react";
 import Head from "next/head";
 import Link from "next/link";
 import styled from "styled-components";
-import Footer from "./Footer";
```
This file is the Main Layout for the whole app, if you remove the `Footer` component from this file it will not render on all the pages.

ZumDeWald commented on components/layout/Layout.tsx:
```diff
@@ -159,8 +158,7 @@ const Layout = ({ children, toggleTheme, isDarkTheme }: LayoutProps) => {
       </Wrapper>
 
       <Main>{children}</Main>
-      <Footer />
```
So here, we need the `Footer` component to stay so the app will render it on every page. The `Main` tag above is where the main body of the app gets dynamically put in/changed.

ZumDeWald commented on components/layout/Layout.tsx:
```diff
@@ -159,8 +158,7 @@ const Layout = ({ children, toggleTheme, isDarkTheme }: LayoutProps) => {
       <Main>{children}</Main>
-      <Footer />
-    </div>
+    </header>
```
This too will need to go back to being an enclosing `div`.

ZumDeWald commented on components/layout/Layout.tsx:
```diff
@@ -95,7 +94,7 @@ const Layout = ({ children, toggleTheme, isDarkTheme }: LayoutProps) => {
 
   return (
-    <div>
+    <header>
```
The `div` here is a container for the whole app, so it will have to stay as an enclosing `div`.

The top section (inside `Wrapper`) is indeed the header, then inside the `Main` tags is where the main part of the app get dynamically rendered, and finally the `Footer` component at the bottom tells Next to render the Footer on every page.

The `header` change would be perfect inside the `Wrapper` component. 
`components > layout > styles.ts` line 4. The `Wrapper` would just need to be based on a `header` instead of a `div`

ZumDeWald reviewed (changes requested):
The changes are a good thing, just not in the best places yet. Lots of comments from me but they are just there to try and be clear on my feedback.

Ibinola committed:
Made some changes

Changed the base element type for the Footerwrapper in style.ts file from a div tag to a footer tag

Ibinola committed:
Made some more changes

Added Import Footer from the Footer component

Ibinola committed:
Made some other changes
